uefi-test-runner/src/boot/misc.rs

```diff
@@ -80,8 +80,7 @@ fn test_watchdog(bt: &BootServices) {
 }
 
 /// Dummy protocol for tests
-#[unsafe_guid("1a972918-3f69-4b5d-8cb4-cece2309c7f5")]
-#[derive(Protocol)]
+#[unsafe_protocol("1a972918-3f69-4b5d-8cb4-cece2309c7f5")]
```

Comment:
general question: why does it need to be called `unsafe`, tho? Why can't the derive attribute be called `uefi_protocol("...")`? I do not see a benefit of the name unsafe here. 

@nicholasbishop

Reply:
If it's called `unsafe_protocol` my question would be what makes it unsafe / how do I use it safely?

Reply:
It's prefixed with `unsafe_` because passing in the wrong GUID can easily lead to unsoundness.

Here's a quick example of how bad things can happen:

The definition of `EFI_DISK_IO_PROTOCOL` from the spec is:
```c
#define EFI_DISK_IO_PROTOCOL_GUID \
    {0xCE345171,0xBA0B,0x11d2,\
    {0x8e,0x4F,0x00,0xa0,0xc9,0x69,0x72,0x3b}}

typedef struct _EFI_DISK_IO_PROTOCOL {
    UINT64 Revision;
    EFI_DISK_READ ReadDisk;
    EFI_DISK_WRITE WriteDisk;
} EFI_DISK_IO_PROTOCOL;
```

Now consider this Rust code:
```rust
#[repr(C)]
// Bad!
#[unsafe_protocol("ce345171-ba0b-11d2-8e4f-00a0c969723b")]
pub struct MyCoolType {
    do_it: extern "efiapi" fn(),
}

let handle = bt.get_handle_for_protocol::<MyCoolType>().unwrap();
let c = bt.open_protocol_exclusive::<MyCoolType>(handle).unwrap();
(c.do_it)();
```

I've defined a protocol called `MyCoolType` with the same GUID as `EFI_DISK_IO_PROTOCOL`. When I call `get_handle_for_protocol` and `open_protocol_exclusive`, UEFI will think it's looking for a disk handle and opening the disk IO protocol. So those calls will succeed, and although `c` will be a `ScopedProtocol<MyCoolType>`, it will contain completely invalid data.

The docstring has a `# Safety` section: https://github.com/rust-osdev/uefi-rs/pull/607/files#diff-ff7f936d7319fc633630baca626659855f359b7185c91237bc9d68bd798256aeR33. There's not much too it; you have to make sure the type and GUID are correct and match up. Definitely open to improving or expanding that paragraph if you have suggestions though.

Reply:
Alright, thanks for the clarification.

Reply:
Thanks!